kernel/context.mli

```diff
@@ -296,11 +296,20 @@ sig
   (** Turn all [LocalDef] into [LocalAssum], leave [LocalAssum] unchanged. *)
   val drop_bodies : ('c, 't) pt -> ('c, 't) pt
 
-  (** [to_instance Ω] builds an instance [args] such
+  (** [to_instance Ω] builds an instance [args] in reverse order such
       that [Ω ⊢ args:Ω] where [Ω] is a named-context and with the local
       definitions of [Ω] skipped. Example: for [id1:T,id2:=c,id3:U], it
       gives [Var id1, Var id3]. All [idj] are supposed distinct. *)
   val to_instance : (Id.t -> 'r) -> ('c, 't) pt -> 'r list
+    [@@ocaml.deprecated "[to_instance] was missing a [List.rev] to comply to its specification; rely on [instance] for the correct specification or use [List.rev (instance ...)] for strict compatibility"]
```

Comment:
Why keep it? Just delete.

Reply:
I'm ok to delete it. It is more that I thought that the policy was to have a deprecation phase as much as possible.

Reply:
IMO deprecation for ML functions that are unused in the CI is overblown. Given that we do not promise any backwards compatibility at the plugin level, it's not very useful and only delays the obvious.

Reply:
I tried to remove it and it happens to be used in equations (with its `List.rev`). I'd prefer to preserve some compatibility for a while. That avoids yet another overlay and that documents what to do to replace it.

Reply:
Generally I think plugin overlays should be done early unless it's trivial but annoying (typically renamings).
But it's your PR so we can merge as-is if you prefer.

Reply:
I just realize that this PR could benefit from a further cleaning: on `rel_context`, the similar functions are named `to_extended_list` and `to_extended_vect`. The two terminologies `to_extended` and `instance` seems as good to me, so I wouldn't object at keeping both as synonyms, but having the same terminology for both `rel` and `named` would be good also. So, I propose to extend the PR with:
- `Context.Rel.instance : (int -> 'r) -> int -> ('c, 't) pt -> 'r array` (alias of `to_extended_vect`)
- `Context.Rel.instance_list : (int -> 'r) -> int -> ('c, 't) pt -> 'r list` (alias of `to_extended_list`)
- `Context.Named.instance_list : (Id.t -> 'r) -> ('c, 't) pt -> 'r list` (in addition to `Context.Named.instance` which is on `array`)

Or we could have:
- `Context.Rel.instance_vect : (int -> 'r) -> int -> ('c, 't) pt -> 'r array` (alias of `to_extended_vect`)
- `Context.Rel.instance_list : (int -> 'r) -> int -> ('c, 't) pt -> 'r list` (alias of `to_extended_list`)
- `Context.Named.instance_list : (Id.t -> 'r) -> ('c, 't) pt -> 'r list`
- `Context.Named.instance_vect : (Id.t -> 'r) -> ('c, 't) pt -> 'r list` (instead of `Context.Named.instance`)

Or, to reuse existing `to_extended` terminology:
- `Context.Rel.to_extended_vect : (int -> 'r) -> int -> ('c, 't) pt -> 'r array` (unchanged)
- `Context.Rel.to_extended_list : (int -> 'r) -> int -> ('c, 't) pt -> 'r list` (unchanged)
- `Context.Named.to_extended_list : (Id.t -> 'r) -> ('c, 't) pt -> 'r list`
- `Context.Named.to_extended_vect : (Id.t -> 'r) -> ('c, 't) pt -> 'r list` (instead of `Context.Named.instance`)

Any opinion?

(By default, I wonder whether the last proposition wouldn't be the simplest, even if the terminology `to_extended` is not necessarily very obvious. Secondarily, I would go to option 1 with synonyms.)

> Generally I think plugin overlays should be done early

Since equations is anyway already tight to master, I'll be ok to do here an overlay.